### ISSUE TITLE
overrides don't recognize dotfiles

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#11201]
**Tell us about your environment**

* **ESLint Version:** 5.10.0
* **Node Version:** 11.1.0
* **npm Version:** 6.5.0

**What parser (default, Babel-ESLint, etc.) are you using?**

- N/A

**Please show your full configuration:**

```yml
root: true

overrides:
  - files: "*.ts"
    parser: typescript-eslint-parser
```

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```bash
eslint --print-config .test.ts
```

**What did you expect to happen?**

It prints:

```json
{
  "globals": {},
  "env": {},
  "rules": {},
  "parserOptions": {},
  "root": true,
  "files": "*.ts",
  "parser": "typescript-eslint-parser"
}
```

**What actually happened? Please include the actual, raw output from ESLint.**

It printed:

```json
{
  "globals": {},
  "env": {},
  "rules": {},
  "parserOptions": {},
  "root": true
}
```

**Are you willing to submit a pull request to fix this bug?**

I'm sorry, I don't have enough time for now.

**Comment by kaicataldo** [comment]
As @Aladdin-ADD mentioned, it looks like the `files` key is currently expected to be an array (see https://github.com/eslint/eslint/blob/5d451c510c15abc41b5bb14b4955a7db96aeb100/lib/config.js and https://github.com/eslint/eslint/blob/caeb223c4f7b0b6fe35e5348ae0df4c6446b5bed/lib/config/config-ops.js). It also looks like the JSDoc for `configOps.pathMatchesGlobs` is incorrect.

I think we should fix `configOps.pathMatchesGlobs` to allow for a single string instead of an array (since it looks like that was the intention), but I also think we should be validating the `files` key of objects in the `overrides` array so that it doesn’t silently fail when misconfigured. I’m willing to do the work if we decide to do this. It’s been a while since I looked at the configuration code and I’d like to familiarize myself again.

**Comment by mysticatea** [comment]
I had written that the code block in "What did you expect to happen?" section is the result of `eslint --print-config test.ts` (note the dot at the beggining of the filename doesn't exist). So, this means, I expected the following two commands to generate the same config.

```
eslint --print-config test.ts
eslint --print-config .test.ts
```
